packages/manager/modules/pci/src/projects/project/storages/containers/containers.service.js

```diff
@@ -132,20 +135,34 @@ export default class PciStoragesContainersService {
       queryParams.archive = false;
     }
 
+    queryParams.region = OPENIO_DEFAULT_REGION;
+
     return this.OvhApiCloudProjectStorage.Aapi()
       .query(queryParams)
       .$promise.then((containers) =>
-        map(containers, (container) => new Container(container)),
+        map(
+          containers,
+          (container) =>
+            new Container({ ...container, publicState: container.public }),
+        ),
       );
   }
 
-  getContainer(projectId, containerId) {
-    return this.OvhApiCloudProjectStorage.v6()
-      .get({
+  getContainer(projectId, containerId, isHighPerfStorage) {
+    if (isHighPerfStorage) {
+      this.promise = this.$http
```

Comment:
do you need to store the promise in Object? if not use can use a constant within the method